### PR TITLE
[WFCORE-4058] Automatic detection of file based KeyStore types

### PR DESCRIPTION
**Pull request by nekdozjam**
https://issues.jboss.org/browse/WFCORE-4058

requires ELY-1650
https://issues.jboss.org/browse/ELY-1650

### CODE REVIEW

File: elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreDefinition.java [hunk 1]
```diff
@@ -82,9 +82,10 @@
 
     static final ServiceUtil<KeyStore> KEY_STORE_UTIL = ServiceUtil.newInstance(KEY_STORE_RUNTIME_CAPABILITY, ElytronDescriptionConstants.KEY_STORE, KeyStore.class);
 
-    static final SimpleAttributeDefinition TYPE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.TYPE, ModelType.STRING, false)
+    static final SimpleAttributeDefinition TYPE = new SimpleAttributeDefinitionBuilder(ElytronDescriptionConstants.TYPE, ModelType.STRING, true)
         .setAttributeGroup(ElytronDescriptionConstants.IMPLEMENTATION)
         .setAllowExpression(true)
+        .setDefaultValue(new ModelNode(ElytronDescriptionConstants.AUTOMATIC))
```
Comment (mchoma) [inline elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreDefinition.java]: From discussion on proposal, I thought there will be no special type introduced. [1]

[1] https://github.com/wildfly/wildfly-proposals/pull/124#discussion_r212284418

File: elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java [hunk 2]
```diff
@@ -108,6 +109,27 @@ static KeyStoreService createFileBasedKeyStoreService(String provider, String ty
     @Override
     public void start(StartContext startContext) throws StartException {
         try {
+
+            if (ElytronDescriptionConstants.AUTOMATIC.equalsIgnoreCase(type)) {
+                pathResolver = pathResolver();
```
Comment (mchoma) [inline elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java]: Lot of lines from added code is duplication to main code flow. Have to be?

File: elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreDefinition.java (hunk 1 above)
Comment (darranl) [reply elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreDefinition.java]: +1 A main motivator for this is to reduce the configuration required so the attribute should just become optional - if the attribute is undefined that becomes our trigger to start automatically detecting it.

Comment (nekdozjam) [reply elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreDefinition.java]: Sorry for that one, fixed

File: elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java (hunk 2 above)
Comment (nekdozjam) [reply elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java]: cleaned up

File: elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java [hunk 3]
```diff
@@ -54,6 +54,7 @@
 import org.wildfly.security.keystore.AliasFilter;
 import org.wildfly.security.keystore.AtomicLoadKeyStore;
 import org.wildfly.security.keystore.FilteringKeyStore;
+import org.wildfly.security.keystore.KeyStoreUtil;
```
Comment (jamezp) [inline elytron/src/main/java/org/wildfly/extension/elytron/KeyStoreService.java]: This missing utility is causing a compilation error.

**Comment by jmesnil** [comment]
@nekdozjam This PR does not compile in its present state. I'm closing it but do not hesitate to reopen it when the code will in a right state (and the Elytron dependencies are up to date)

**Comment by darranl** [comment]
@nekdozjam Ping me once you are ready to look at this and we will talk about getting it on top of https://github.com/wildfly/wildfly-core/pull/3619 - also we will likely need a transformer for this one.

**Pull request closed without merging**